dms/models/directory.py

```diff
@@ -273,11 +273,14 @@ def _get_parent_categories(self, access_token):
         return directories
 
     def _get_own_root_directories(self):
-        return (
-            self.env["dms.directory"]
-            .search([("is_hidden", "=", False), ("parent_id", "=", False)])
-            .ids
+        res = self.env["dms.directory"].search_read(
+            [("is_hidden", "=", False)], ["display_name", "parent_id"]
         )
+        all_ids = [value["id"] for value in res]
+        for index, item in enumerate(res):
+            if item["parent_id"] and item["parent_id"][0] in all_ids:
```

Comment:
Aren't this keeping subdirectories of third-level?

Reply:
This will delete subdirectories if we have access to the parent directory.

Example:
![ejemplo-1](https://user-images.githubusercontent.com/4117568/217550175-3536b5ae-03ef-4609-8c4c-b2c7664767e6.png)
![ejemplo-2](https://user-images.githubusercontent.com/4117568/217550184-bbad8a12-4662-4839-9679-1299329fa3f7.png)

Reply:
But you can see the problem in your screenshot: "2023 (level 3)" shouldn't be displayed there. Add the previously removed folders in the list / another list for being taking into account in successive removals.

Reply:
Sorry, but what the screenshot shows is what you indicated at https://github.com/OCA/dms/pull/226#discussion_r1098947571, right?

Reply:
True, as "Slides" is not allowed, but please put "2023 (level 3)" inside "Code (level 2)". Only code should be shown, but if I read your code, there may be the possibility of both being shown.